src/NuGet.Core/NuGet.Protocol.Core.Types/UserAgent.cs

```diff
@@ -58,6 +63,11 @@ public static string CreateUserAgentStringForVisualStudio(string client, string
                 throw new ArgumentNullException(nameof(client));
             }
 
+            if (NuGetTestMode.Enabled)
```

Comment:
This ensures that, when the environment variable is set, any tests run using any client will use the nuget test client useragent instead.